### PR TITLE
SOX-14 add new states

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: SOX-14
KBC: https://github.com/keboola/connection/pull/4468 

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)
    *
- testuje par zakladnich usecasu ohledne WF

### CODE REVIEW

File: tests/Backend/SOX/MergeRequestsTest.php [hunk 1]
```diff
@@ -93,6 +93,42 @@ public function testPutInReview(): void
...
+
+        $reviewerClient = $this->getReviewerStorageApiClient();
+        $this->developerClient->mergeRequestPutToReview($mrId);
+
+        $mrData = $reviewerClient->mergeRequestAddApproval($mrId);
+
+        $this->assertEquals('in_review', $mrData['state']);
+        $this->assertCount(1, $mrData['approvals']);
+
+        $mrData = $reviewerClient->mergeRequestAddApproval($mrId);
```
Comment (romanbracinik) [inline tests/Backend/SOX/MergeRequestsTest.php:118]: viem ze to je draft, len som si to vsimol... nemalo by to takto fungovat nie ? ze jednim tokenom dam 2x approve. to by jeden clovek mohol schvalit cely MR. imho to musia byt dva rozne tokeny

Comment (jirkasemmler) [reply tests/Backend/SOX/MergeRequestsTest.php:118]: jo, mas pravdu, uz to je v connection opravene a tady ten test zacne failovat

File: src/Keboola/StorageApi/Client.php [hunk 2]
```diff
@@ -921,6 +921,39 @@ public function mergeRequestPutToReview(int $mergeRequestId): array
         return $mrDetail;
     }
 
+    /**
```
Comment (romanbracinik) [inline src/Keboola/StorageApi/Client.php:924]: tie PHPDoc tu uz nemusia byt ked to mas typovane

File: tests/Backend/SOX/MergeRequestsTest.php [hunk 3]
```diff
@@ -93,6 +93,67 @@ public function testPutInReview(): void
...
+
+        $this->assertEquals('in_review', $mrData['state']);
+        $this->assertCount(1, $mrData['approvals']);
+
+        try {
+            $mrData = $reviewerClient->mergeRequestAddApproval($mrId);
+        } catch (ClientException $e) {
+            $this->assertSame('Operation canot be performed due: This reviewer has already approved this request.', $e->getMessage());
+        }
+    }
```
Comment (romanbracinik) [inline tests/Backend/SOX/MergeRequestsTest.php:155]: Este by som otestoval, ze ked pridam druhy approve tak sa to naozaj approvne

**Review by romanbracinik — CHANGES_REQUESTED** [review]
Nejake drobnosti, a fixni prosim ten stan

File: tests/Backend/SOX/MergeRequestsTest.php (hunk 3 above)
Comment (jirkasemmler) [reply tests/Backend/SOX/MergeRequestsTest.php:155]: jo, na to ale musim predelat generator projektu, protoze jak jsi psal v minulem komentari - je na to potreba dalsi token s REVIEWER roli. udelam si na to dalsi task

Comment (romanbracinik) [reply tests/Backend/SOX/MergeRequestsTest.php:155]: Ok alebo myslis ze by to slo napr zablokovat az v momente toho ked budeme mat dorieseny ten test ? Pretoze vlastne bez toho nemozem napisat testy na svoje mergovanie, pretoze si to nemam ako approvnut :) Alebo predpokladas ze to bude nieco rychle :)

File: tests/Backend/SOX/MergeRequestsTest.php (hunk 1 above)
Comment (jirkasemmler) [reply tests/Backend/SOX/MergeRequestsTest.php]: ha, jeste, ze jsi to pripomnel, protoze jsem mel v connection bug

**Review by romanbracinik — APPROVED** [review]
👌

**Pull request merged**